### DISCUSSION TITLE
DNS resolving not working anymore in Iran!

### DISCUSSION THREAD

**Discussion by persian85** [DNSCrypt/dnscrypt-proxy#2222] [Q&A]
If you followed the news you are aware of the recent internet limitations in Iran. So I cannot resolve DNS requests with dnscrypt-proxy anymore. The service starting without any problem like before and could connect to lots of servers but after requesting any dns queries it doesn't work! All it return is:
facebook.com
;; communications error to 127.0.0.1#53: timed out
;; communications error to 127.0.0.1#53: timed out
;; communications error to 127.0.0.1#53: timed out
;; no servers could be reached
 I looked into its log and it shows it could connect to at least 20 servers but couldnt resolve any queries! 
What can I do about that? Any advice please.

**Comment by jedisct1** [comment]
What does `dnscrypt-proxy -resolve example.com` show?

**Reply by persian85** [reply]
Resolving [facebook.com] using 127.0.0.1 port 53

Unable to resolve: [Timeout]

**Reply by persian85** [reply]
FYI I set the timeout option to maximum 10000.

**Reply by jedisct1** [reply]
How about `dig use-application-dns.net @127.0.0.1` ?

**Reply by persian85** [reply]
; <<>> DiG 9.18.7 <<>> use-application-dns.net @127.0.0.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 36119
;; flags: qr ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;use-application-dns.net.	IN	A

;; Query time: 0 msec
;; SERVER: 127.0.0.1#53(127.0.0.1) (UDP)
;; WHEN: Tue Oct 18 06:58:40 +0330 2022
;; MSG SIZE  rcvd: 52

**Reply by jedisct1** [reply]
Ok, at least the proxy responds to queries.

Are you using Anonymized DNSCrypt?

DoH and ODoH are easy to block, but it would be surprising if they also blocked DNS relays.

**Reply by persian85** [reply]
No I dont.
I also set those features (DoH & ODoH) to false but nothing changed! Service could connect to 35 servers but cannot resolve anything!

**Reply by persian85** [reply]
I did some tune in config file and could make it work with TOR service but the problem is that TOR service is unstable due to the Internet limitations. Any advice?

**Comment by persian85** [comment]
Any idea?

**Reply by jedisct1** [reply]
Did you try Anonymized DNS?

**Reply by persian85** [reply]
Would you please tell me how can I try that?

**Reply by persian85** [reply]
OK I found the Anonymized section on config file and set the line "skip_incompatible = false to True" so now after restarting service it cannot connect to any servers and returns TIMEOUT!

**Reply by jedisct1** [reply]
Just like ODOH, you need to define relays: https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Anonymized-DNS

**Reply by persian85** [reply]
OK so I did like below: 
routes = [   { server_name='*', via=['anon-cs-fr', 'anon-bcn'] } ]
skip_incompatible = true
direct_cert_fallback = fasle
and tested some other anon server but it returns TIMEOUT! below lines is an example of my output:
Oct 24 12:44:24 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:24] [NOTICE] Anonymizing queries for [plan9dns-nj] via [anon-cs-berlin]
Oct 24 12:44:34 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:34] [NOTICE] [plan9dns-nj] TIMEOUT
Oct 24 12:44:34 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:34] [NOTICE] Anonymizing queries for [dnswarden-uncensor-dc] via [anon-cs-berlin]
Oct 24 12:44:45 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:45] [NOTICE] [dnswarden-uncensor-dc] TIMEOUT
Oct 24 12:44:45 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:45] [NOTICE] Anonymizing queries for [serbica] via [anon-cs-berlin]
Oct 24 12:44:55 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:55] [NOTICE] [serbica] TIMEOUT
Oct 24 12:44:55 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:44:55] [NOTICE] Anonymizing queries for [jp.tiar.app] via [anon-cs-berlin]
Oct 24 12:45:06 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:45:06] [NOTICE] [jp.tiar.app] TIMEOUT
Oct 24 12:45:06 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:45:06] [NOTICE] Anonymizing queries for [ibksturm] via [anon-cs-berlin]
Oct 24 12:45:16 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:45:16] [NOTICE] [ibksturm] TIMEOUT
Oct 24 12:45:16 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:45:16] [NOTICE] Anonymizing queries for [sby-limotelu] via [anon-cs-berlin]
Oct 24 12:45:27 archdesk dnscrypt-proxy[450765]: [2022-10-24 12:45:27] [NOTICE] [sby-limotelu] TIMEOUT

:-(

**Reply by persian85** [reply]
After waiting about 1 hour it only could connect to 1 server and even by that cannot reslove any query and retuns TIMEOUT!

**Comment by persian85** [comment]
So this is my report after 1 day of trying to use Anonymized DNS.
I set * for servers and anon servers to try every possible server to make it work. now after 1 day of service trying to connect any servers via any anon servers it just could connect to 3 different servers but still cannot resolve any dns queries and returns TIMEOUT. 
So it seems there should be some changes to be made by maintainer to make it work in such situations like mine in Iran with lots of internet limitations made by government.

**Reply by dapphp** [reply]
@persian85 Thanks for providing this update. Do you use Signal (signal.org)? If we can find a secure way to communicate I'd like to see if we can work together to figure get around this block or figure out the problem.

Maybe they are blocking known DNSCrypt / Anonymous DNS server addresses. In that case, we could try setting up a new one on a fresh IP address that might not be blocked. Link to my contact info is in my Github profile.

**Reply by persian85** [reply]
Thank you. I'll send you a message there.
